### PR TITLE
solve issue #988, add reddit and HN share buttons

### PR DESCRIPTION
**Pull request by iynewz**
image
image
image

### CODE REVIEW

File: src/components/share/hackerNewsShareButton.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  [key: string]: string | number | undefined | null;
+}) {
+  const params = Object.entries(object)
+    .filter(([, value]) => value !== undefined && value !== null)
+    .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(String(value))}`);
+
+  return params.length > 0 ? `?${params.join('&')}` : '';
+}
+
+import React, { Ref, forwardRef } from 'react';
```
Comment (shanghaikid) [inline src/components/share/hackerNewsShareButton.tsx:28]: always put imports on the top of the code.

File: src/components/share/hackerNewsShareButton.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  optsMap: (props: OptionProps) => LinkOptions,
+  defaultProps: Partial<ShareButtonProps<LinkOptions> & OptionProps>,
+) {
+  type Props = Omit<
+    ShareButtonProps<LinkOptions>,
+    'forwardedRef' | 'networkName' | 'networkLink' | 'opts'
+  > &
+    OptionProps;
+  
+  console.log();
```
Comment (shanghaikid) [inline src/components/share/hackerNewsShareButton.tsx:44]: remove all consoles please

File: src/components/share/hackerNewsShareButton.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+function hackerNewsLink(
+  url: string,
+  {
+    title,
+    via,
+    hashtags = [],
+    related = [],
+  }: { title?: string; via?: string; hashtags?: string[]; related?: string[] },
+) {
+  assert(url, 'hackernews.url');
```
Comment (shanghaikid) [inline src/components/share/hackerNewsShareButton.tsx:93]: please remove these testing function calls in the code.

**Review by shanghaikid — CHANGES_REQUESTED** [review]

**Comment by iynewz** [comment]
please take a look again @shanghaikid

**Review by shanghaikid — APPROVED** [review]

**Pull request merged**